File: ant-sdk/antd-php/src/Models/PrepareUploadResult.php
<?php

declare(strict_types=1);

namespace Autonomi\Antd\Models;

/**
 * Result of preparing an upload (file or in-memory data) for external signing.
 *
 * Returned by AntdClient::prepareUpload() and AntdClient::prepareDataUpload()
 * (and their Async variants). Wave-batch shape only — merkle support can be
 * added later if/when needed.
 */
readonly class PrepareUploadResult
{
    public function __construct(
        /** Opaque server-side identifier to pass back to finalizeUpload(). */
        public string $uploadId,
        /** "wave_batch" for the only mode currently exposed by this PHP client. */
        public string $paymentType,
        /** @var list<PaymentInfo> Per-quote payments the external signer must submit. */
        public array $payments,
        /** Total atto-token amount across all entries in $payments. */
        public string $totalAmount,
        /** Payment vault contract address (hex with 0x prefix). */
        public string $paymentVaultAddress,
        /** Payment token contract address (hex with 0x prefix). */
        public string $paymentTokenAddress,
        /** EVM RPC URL the external signer should submit transactions through. */
        public string $rpcUrl,
        /**
         * Total chunks in this upload, including any already on-network.
         * Added in antd 0.10.0; 0 against older daemons. The external signer
         * pays for ($totalChunks - $alreadyStoredCount) chunks.
         */
        public int $totalChunks = 0,
        /** Chunks already stored on-network and excluded from payment + PUT (added in antd 0.10.0). */
        public int $alreadyStoredCount = 0,
    ) {}
}

File: ant-sdk/antd-java/src/main/java/com/autonomi/antd/models/PrepareUploadResult.java
package com.autonomi.antd.models;

import java.util.List;

/**
 * Result of preparing an upload for external signing.
 * PaymentType is "wave_batch" or "merkle" — determines which fields are populated
 * and which contract call the external signer must make.
 *
 * @param uploadId               hex identifier for this upload session
 * @param paymentType            "wave_batch" or "merkle"
 * @param payments               per-quote payments for payForQuotes() (wave_batch only)
 * @param totalAmount            total atto tokens ("0" for merkle)
 * @param paymentVaultAddress     payment vault contract address
 * @param paymentTokenAddress    payment token contract address
 * @param rpcUrl                 EVM RPC URL for submitting transactions
 * @param depth                  merkle tree depth 1-8 (merkle only)
 * @param poolCommitments        pool commitments for payForMerkleTree() (merkle only)
 * @param merklePaymentTimestamp unix seconds timestamp (merkle only)
 * @param totalChunks            total chunks incl. already-stored (added in antd 0.10.0; 0 on older daemons)
 * @param alreadyStoredCount     chunks already on-network, excluded from payment (added in antd 0.10.0)
 */
public record PrepareUploadResult(
        String uploadId,
        String paymentType,
        List<PaymentInfo> payments,
        String totalAmount,
        String paymentVaultAddress,
        String paymentTokenAddress,
        String rpcUrl,
        Integer depth,
        List<PoolCommitmentEntry> poolCommitments,
        Long merklePaymentTimestamp,
        long totalChunks,
        long alreadyStoredCount) {}
